### PR TITLE
Work/load translations early

### LINKED ISSUE #11142: Setup translations before `Application` instantiation
The version checks have translatable text, and are shown before the translations are set up. Move this to somewhere early in the `main()` function.

### LINKED ISSUE #11246: Deprecate legacy `--confdir` command-line  argument
This command-line argument causes problems with the language setup, see issue #11142 .

### PR DESCRIPTION
**Pull request by erikjv**
Remove the `--confdir` option and load translations before any messages are shown.

Fixes #11246. Fixes #11142.

Waits on #11316.

### CODE REVIEW

**Comment by erikjv** [comment]
This still needs changelog entries

**Comment by erikjv** [comment]
> This still needs changelog entries

Done.

File: src/gui/main.cpp [hunk 1]
```diff
@@ -114,25 +116,26 @@ CommandLineOptions parseOptions(const QStringList &arguments)
         return option;
     };
 
-    auto showSettingsLegacyOption = QCommandLineOption{{QStringLiteral("showsettings")}, QStringLiteral("Hidden legacy option")};
+    auto showSettingsLegacyOption = QCommandLineOption{{QStringLiteral("showsettings")}, QApplication::translate("CommandLine", "Hidden legacy option")};
```
Comment (TheOneRing) [inline src/gui/main.cpp:119]: As it is hidden I don't think we need to translate it

**Comment by TheOneRing** [comment]
@saw-jan I just realised the gui tests are relying on `--confdir` could you elaborate why?

**Comment by saw-jan** [comment]
> @saw-jan I just realised the gui tests are relying on `--confdir` could you elaborate why?

yeah, we are using that option to use different config dir other than default one so that we don't mess up configs while running tests locally.

Fabian suggested if we can use `XDG_CONFIG_HOME` env during tests. so, we are going to try that one

**Comment by TheOneRing** [comment]
> > @saw-jan I just realised the gui tests are relying on `--confdir` could you elaborate why?
> 
> yeah, we are using that option to use different config dir other than default one so that we don't mess up configs while running tests locally.
> 
> Fabian suggested if we can use `XDG_CONFIG_HOME` env during tests. so, we are going to try that one

thx can you prepare a pr? (please don't commit to this one)

**Comment by saw-jan** [comment]
>thx can you prepare a pr? (please don't commit to this one)
 yeah, on the way

**Comment by saw-jan** [comment]
>yeah, on the way

PR is ready to be merged: https://github.com/owncloud/client/pull/11316

File: src/gui/main.cpp (hunk 1 above)
Comment (erikjv) [reply src/gui/main.cpp]: done

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**